GameServerLib/ObjectManager.cs

```diff
@@ -135,9 +135,20 @@ public void Update(float diff)
                 if (ai != null)
                 {
                     var tempBuffs = ai.GetBuffs();
-                    foreach (var buff in tempBuffs.Values)
-                    {                       
-                        buff.Update(diff);
+                    for (int i = 0; i < tempBuffs.Count; i++)
+                    {
+                        var tempbuff = tempBuffs[i];
+                        if (tempbuff.Elapsed())
+                        {
+                            if (ai.HasBuff(tempbuff))
```

Comment:
What are the scenarios that `ai` doesn't have the buff? Also, please merge the `ifs` into one `if`. ( `if (tempbuff.Elapsed() && ai.HasBuff(tempbuff))` if it works)

The code in the loop seems to be a little bit of spaghetti (some `continue` and ifs that most of the time never occur and correlated with others).

Reply:
The continue is necessary as it prevents elapsed buffs from Updating.